src/orchestrator.rs

```diff
@@ -220,18 +235,22 @@ async fn detect(
             let ctx = ctx.clone();
             let detector_id = detector_id.clone();
             let detector_params = detector_params.clone();
-            let chunker_id = ctx.config.get_chunker_id(&detector_id);
+            let chunker_id =
+                ctx.config
+                    .get_chunker_id(&detector_id)
+                    .ok_or_else(|| Error::DetectorNotFound {
```

Comment:
Yes, it will be checked during config validation, so in theory this error should never be returned here. We should still be safe and use `ok_or_else()` here rather than `unwrap()` though.